monitoring/space_monitor.py

```diff
@@ -0,0 +1,251 @@
+        daemonize.daemonize()
+    start_logging(args.logfile, args.loglevel)
+    notify = choose_notify_function(args.notify, emails, mail_config)
+    check_periodically(args.interval, args.reminder_days,
+                       notify, datasets, users)
+
+
+def parse_arguments():
+    parser = ArgumentParser(description=__doc__, formatter_class=fmt)
+    add = parser.add_argument  # shortcut
+    add('--no-daemon', action='store_true', help='keep attached to the console')
+    add('--logfile', default=LOGFILE, help='path of the file with the logs')
+    add('--loglevel', choices=['error', 'warning', 'info', 'debug'],
+        default='warning', help='logger verbosity level')
+    add('--reminder-days', default=7, help='number of days between reminders')
+    add('--notify', choices=['email', 'print'], default='email',
+        help='method used for notification')
+    add('--interval', type=int, default=60, help='time (in s) between checks')
+    add('--config', default='space_monitor.cfg',
+        help='file with auth, datasets and users configuration')
+    return parser.parse_args()
+
+
+def read_config(fname):
+    "Return parameters read from configuration file fname"
+    print('Reading config file %s ...' % fname)
+    cp = ConfigParser()
+    try:
+        cp.read_file(open(fname))
+        for section in ['mail', 'datasets', 'users']:
+            assert section in cp, 'Missing section [%s]' % section
+        for x in ['from', 'password', 'cc']:
+            assert x in cp['mail'], 'Missing field "%s" in section [mail]' % x
+    except (FileNotFoundError, AssertionError,
+            ValueError, ParsingError) as e:
+        sys.exit('Error in file %s: %s' % (fname, e))
+
+    mail_config = cp['mail']
+    datasets = list(cp['datasets'].keys())
+    users = list(cp['users'].keys())
+    emails = {space: emails.split() for space, emails in
+              list(cp['datasets'].items()) + list(cp['users'].items())}
+    return mail_config, datasets, users, emails
+
+
+def start_logging(logfile, loglevel):
+    level = {'error': 40, 'warning': 30, 'info': 20, 'debug': 10}[loglevel]
+    logging.basicConfig(filename=logfile, level=level,
+                        format='%(asctime)s [%(levelname)s] %(message)s')
+    logging.info('Starting.')
+    logging.debug('Current directory is %s' % os.getcwd())
+
+
+def choose_notify_function(method, emails, mail_config):
+    "Return a function used to notify"
+    if method == 'email':
+        def notify_by_email(space, status_new, status_old):
+            subject = 'Space at %s: %s' % (os.uname().nodename, status_new)
+            text = describe(space, status_new, status_old)
+            send_email(emails[space], subject, text, mail_config)
+        return notify_by_email
+    elif method == 'print':
+        return print  # the print function (useful for debugging)
+
+
+def check_periodically(interval, reminder_days, notify, datasets, users):
```

Comment:
Idea: How about making the script runnable from a crontab instead of daemonizing it + infinite loop? `warnquota` works this way. It simplifies both the script and the deploy, it's simply a script that does the job and exists. Would this work for us?

Reply:
It works great!